gateway-ha/src/main/java/io/trino/gateway/ha/router/RoutingManager.java

```diff
@@ -185,6 +225,12 @@ protected String findBackendForUnknownQueryId(String queryId)
         return gatewayBackendManager.getActiveAdhocBackends().get(0).getProxyTo();
     }
 
+    protected String lookupBackendForCookie(String cookie)
```

Comment:
hm... whats the difference to findBackend? not clear from name

Reply:
the difference was that this method looks up the cookie in the db, while findBackend gets it from the local cache. However this was only used in the cache builder so it makes more sense just to inline it.